### PR TITLE
[GSoC] Body zero inertia

### LINKED ISSUE #23269: KanesMethod cannot distinguish if a Body is Particle or RigidBody
I am using `sympy==1.10` with `python==3.9`.

I am deriving the equation of motion of my system using KanesMethod and Body objects, and some of them represent `Particle`s and the others `RigidBody`:

```
particle = Body('p', mass=..., frame=..., masscenter=...)
body = Body('b', mass=..., frame=..., masscenter=..., central_inertia=...)

KM = KanesMethod(...)
KM.kanes_equations([particle, body], [...])
```

But the problem is, [`KanesMethod._form_frstar()`](https://github.com/sympy/sympy/blob/37c7a4e7599ae60db3e12012097c175777db2254/sympy/physics/mechanics/kane.py#L338) tries to refer to `body.central_inertia` even though the `body` is a `Particle`. 

It is because the function uses [`isinstance(body, RigidBody)`](https://github.com/sympy/sympy/blob/37c7a4e7599ae60db3e12012097c175777db2254/sympy/physics/mechanics/kane.py#L382) to determine if a `body` is an instance of `RigidBody` or `Particle`. Since [`Body`](https://github.com/sympy/sympy/blob/37c7a4e7599ae60db3e12012097c175777db2254/sympy/physics/mechanics/body.py#L10) class itself inherits both `RigidBody` and `Particle`, both `isinstance(body, RigidBody)` and `isinstance(body, Particle)` are always `true` no matter if it means to represent a `Particle` or `RigidBody`.
(It seems that the same code is being used in [`get_partial_velocity()`](https://github.com/sympy/sympy/blob/37c7a4e7599ae60db3e12012097c175777db2254/sympy/physics/mechanics/kane.py#L362), too.)

Could you review the issue? If I am using the class in a wrong way, please let me know!

### PR DESCRIPTION
**Pull request by tjstienstra**
#### References to other Issues or PRs

Partially fixes #23269

#### Brief description of what is fixed or changed
Change the inertia of a Body, which represents as Particle, to zero. And add a central inertia setter to RigidBody.

#### Release Notes

* physics.mechanics
  * A Body representing a particle now has a zero inertia.
  * Implemented a central inertia setter.

### CODE REVIEW

File: sympy/physics/mechanics/body.py [hunk 1]
```diff
@@ -156,9 +156,19 @@ def z(self):
         """The basis Vector for the Body, in the z direction. """
         return self.frame.z
 
+    @property
+    def inertia(self):
```
Comment (moorepants) [inline sympy/physics/mechanics/body.py:160]: A simple docstring is needed for these properties. Should the be called central_inertia?

File: sympy/physics/mechanics/tests/test_body.py [hunk 2]
```diff
@@ -62,6 +62,8 @@ def test_particle_body():
     assert hasattr(particle_body, 'frame')
     assert hasattr(particle_body, 'masscenter')
     assert hasattr(particle_body, 'mass')
+    assert particle_body.inertia == (Dyadic(0), particle_body.masscenter)
+    assert particle_body.central_inertia == Dyadic(0)
```
Comment (moorepants) [inline sympy/physics/mechanics/tests/test_body.py:66]: It seems `.inertia` gives the central inertia, so why are both needed here? What if you want an inertia about another point? Is that part of the API?

File: sympy/physics/mechanics/body.py [hunk 3]
```diff
@@ -156,9 +156,19 @@ def z(self):
...
+        if self.is_rigidbody:
+            return RigidBody.inertia.fget(self)
+        return (self.central_inertia, self.masscenter)
+
+    @inertia.setter
+    def inertia(self, I):
+        RigidBody.inertia.fset(self, I)
+
```
Comment (moorepants) [inline sympy/physics/mechanics/body.py:168]: i didn't see any tests for these additions to Body.

Comment (moorepants) [reply sympy/physics/mechanics/body.py:168]: nevermind this, I now see them

File: sympy/physics/mechanics/tests/test_body.py (hunk 2 above)
Comment (moorepants) [reply sympy/physics/mechanics/tests/test_body.py:66]: I see now that RigidBody can be defined with its `.inertia` wrt to any point. Does this work for particle too? A particle can have inertia wrt a point other than its mass center.

File: sympy/physics/mechanics/body.py (hunk 1 above)
Comment (TJStienstra) [reply sympy/physics/mechanics/body.py:160]: No, `inertia` is overall defined as being a `tuple` `(Dyadic, Point)` (see `RigidBody`). However personally I would just most of the time set the central inertia of a body. The argument for `Body` also implies this preference, as it is `central_inertia` and not `inertia` like for `RigidBody`. This is also the reason why I've added a `central_inertia.setter` to `RigidBody`.

File: sympy/physics/mechanics/tests/test_body.py (hunk 2 above)
Comment (TJStienstra) [reply sympy/physics/mechanics/tests/test_body.py:66]: I've just added a test showing the expected behaviour.

File: sympy/physics/mechanics/tests/test_body.py [hunk 4]
```diff
@@ -62,8 +62,22 @@ def test_particle_body():
...
+    assert particle_body.central_inertia == inertia(particle_frame, 1, 1, 1)
+    assert particle_body.is_rigidbody
 
+    particle_body = Body('particle_body', mass=particle_mass)
     assert not particle_body.is_rigidbody
+    point = particle_body.masscenter.locatenew('point', particle_body.x)
+    point_inertia = particle_mass * inertia(particle_body.frame, 0, 1, 1)
+    particle_body.inertia = (point_inertia, point)
+    assert particle_body.inertia == (point_inertia, point)
+    assert particle_body.central_inertia == Dyadic(0)
```
Comment (moorepants) [inline sympy/physics/mechanics/tests/test_body.py:79]: I'm confused about this. It seems that you set the inertia of this body to be Ixx=0, Iyy=1, Izz=1 about a point 1 unit along x from the mass center. Then you ask for the inertia about the mass center here and it gives zero. If the body has inertia about y and z, shouldn't this be nonzero? Or is it that the mass=1 and r=1, so m*r**2=1, then you substract 1 from 1 for the y and z terms. I guess that is it.

Comment (moorepants) [reply sympy/physics/mechanics/tests/test_body.py:79]: I convinced myself there.

**Comment by moorepants** [comment]
This LGTM!

**Pull request merged**